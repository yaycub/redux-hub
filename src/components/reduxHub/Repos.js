import React from 'react';
import { useSelector } from 'react-redux';
import { selectRepoState, selectRepoLoadingState } from '../../selectors/selectRepoState';
import Repo from './Repo';

const Repos = () => {
  const repos = useSelector(selectRepoState);
  const loading = useSelector(selectRepoLoadingState);

  const reposEls = repos.map(repo => (
    <li key={repo.name}>
      <Repo {...repo} />
    </li>
  ));

  if(loading) <p></p>;

  return (
    <ul>
      {reposEls}
    </ul>
  );
};

export default Repos;
