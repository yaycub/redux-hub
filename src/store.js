import { createStore } from 'redux';
import thunk from 'redux-thunk';



