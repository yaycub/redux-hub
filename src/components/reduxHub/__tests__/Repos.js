import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../store';
import Repos from '../Repos';

describe('Search Form Component', () => {
  it('should render Search Form', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Repos />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
