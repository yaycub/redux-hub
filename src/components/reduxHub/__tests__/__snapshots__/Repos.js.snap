// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form Component should render Search Form 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Repos>
    <ul />
  </Repos>
</Provider>
`;
